### PR TITLE
[NodeJS] Redesigned the structure of AzureEnvironment and updated environment info

### PR DESCRIPTION
**Pull request by amarzavery**
- https://github.com/Azure/azure-sdk-for-node/issues/1707

### CODE REVIEW

**Comment by amarzavery** [comment]
@begoldsm, @yugangw-msft - Can you review this PR? This PR restructures AzureEnvironment.js

**Comment by begoldsm** [comment]
@amarzavery looking now.

File: ClientRuntimes/NodeJS/ms-rest-azure/lib/azureEnvironment.js [hunk 1]
```diff
@@ -2,48 +2,158 @@
...
+    activeDirectoryResourceId: 'https://management.core.cloudapi.de/',
+    activeDirectoryGraphResourceId: 'https://graph.cloudapi.de/',
+    activeDirectoryGraphApiVersion: '2013-04-05',
+    storageEndpointSuffix: '.core.cloudapi.de',
+    keyVaultDnsSuffix: '.vault.microsoftazure.de',
+    // TODO: add dns suffixes for the US government for datalake store and datalake analytics once they are defined.
+    azureDataLakeStoreFileSystemEndpointSuffix: 'N/A',
+    azureDataLakeAnalyticsCatalogAndJobEndpointSuffix: 'N/A'
+  }
+};
```
Comment (begoldsm) [inline ClientRuntimes/NodeJS/ms-rest-azure/lib/azureEnvironment.js]: Question: Does it make sense to include the dogfood environment here, or do we not want to expose those URIs in the code?

**Comment by begoldsm** [comment]
@amarzavery looks good. Do you need to also rev the version of the ms-rest-azure package? I didn't see the package.json in the review.

File: ClientRuntimes/NodeJS/ms-rest-azure/lib/azureEnvironment.js (hunk 1 above)
Comment (amarzavery) [inline ClientRuntimes/NodeJS/ms-rest-azure/lib/azureEnvironment.js]: Dont want to expose the internal environments here. Have provided a mechanism to add new environment.

**Comment by amarzavery** [comment]
@begoldsm - Thanks for reminding. Shall update that.

**Pull request merged**